CONTRIBUTING.md

```diff
@@ -2,26 +2,70 @@
 
 Welcome! As a [Jupyter](https://jupyter.org) project, we follow the [Jupyter contributor guide](https://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html).
 
-## Local deployment
+## Minikube
 
 To develop binderhub, you can use a local installation of JupyterHub on minikube,
 and run binderhub on the host system.
 
-1. Follow [zero-to-jupyterhub](https://zero-to-jupyterhub.readthedocs.io/en/latest/setup-jupyterhub.html#install-jupyterhub) to setup minikube
+1. [Install Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/).
 
-2. deploy JupyterHub with helm
+2. Install helm
+
+   ```bash
+   curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash
+   ```
+
+   [Alternative methods](https://docs.helm.sh/using_helm/#installing-the-helm-client) for helm installation
+   exist if you prefer installing without using the script.
+
+3. Initialize helm in minikube
+
+   ```bash
+   helm init
+   ```
+4. Add the JupyterHub helm charts repo:
+
+   ```bash
+   helm repo add jupyterhub https://jupyterhub.github.io/helm-chart/
+   helm repo update
+   ```
+
+5. Install JupyterHub in minikube with helm
 
         helm install jupyterhub/jupyterhub --version=v0.5.x-2140cd6 \
             --name=binder \
             --namespace=binder \
-            -f testing/minikube-config.yaml
+            -f testing/minikube/jupyterhub-helm-config.yaml
 
-3. install binderhub:
+6. Install binderhub:
 
         python3 -m pip install -e .
 
-4. start binderhub with the testing config file:
+7. Start binderhub with the testing config file:
+
+        python3 -m binderhub -f testing/minikube/binderhub_config.py
+
+8. Visit [http://localhost:8585](http://localhost:8585)
+
+All features should work, including building and launching.
+
+## Pure HTML / CSS / JS development
+
+If you do not want to set up minikube but just want to hack on the html / css / js,
+there is a simpler method!
+
+1. Install binderhub:
+
+   ```bash
+   python3 -m pip install -e .
+   ```
+
+2. Run it!
+
+   ```bash
+   python3 -m binderhub.app -f testing/localonly/binderhub_config.py
+   ```
 
-        python3 -m binderhub -f testing/binderhub_config.py
+3. You can now access it locally at https://localhost:8585
```

Comment:
Running locally works for view css/html but there is an error thrown when there is not a valid GH repo on start.
```
$ python3 -m binderhub.app -f testing/localonly/binderhub_config.py --debug
[BinderHub] Looking for testing/localonly/binderhub_config in /Users/carol/code/jhub/binderhub
[BinderHub] Loaded config file: /Users/carol/code/jhub/binderhub/testing/localonly/binderhub_config.py
[I 171007 00:00:22 app:261] BinderHub starting on port 8585
[I 171007 00:00:36 web:2063] 304 GET / (::1) 18.64ms
[E 171007 00:00:40 builder:120] Failed to get provider for gh:/master
    Traceback (most recent call last):
      File "/Users/carol/code/jhub/binderhub/binderhub/builder.py", line 118, in get
        provider = self.get_provider(provider_prefix, spec=spec)
      File "/Users/carol/code/jhub/binderhub/binderhub/base.py", line 19, in get_provider
        config=self.settings['traitlets_config'], spec=spec)
      File "/Users/carol/code/jhub/binderhub/binderhub/repoproviders.py", line 107, in __init__
        raise ValueError(msg)
    ValueError: Spec is not of the form "user/repo/ref", provided: "/master".
[I 171007 00:00:40 web:2063] 200 GET /build/gh//master (::1) 2.27ms
```